### PR TITLE
Fix ICE due to mishandling of Aggregate rvalue for raw pointers to trait objects

### LINKED ISSUE #3631: Unexpected kani panic: assign statement with unequal types
This issue is related to the [NonNull module verification](https://github.com/model-checking/verify-rust-std/discussions/124).

This is a POC:
```rust
#![feature(ptr_metadata)]

use std::ptr::NonNull;

trait SampleTrait {
    fn get_value(&self) -> i32;
}

struct SampleStruct {
    value: i32,
}

impl SampleTrait for SampleStruct {
    fn get_value(&self) -> i32 {
        self.value
    }
}

#[cfg(kani)]
#[kani::proof]
fn main() {

    // Create a SampleTrait object from SampleStruct
    let sample_struct = SampleStruct { value: kani::any() };
    let trait_object: &dyn SampleTrait = &sample_struct;

    // Get the raw data pointer and metadata for the trait object
    let trait_ptr = NonNull::new(trait_object as *const dyn SampleTrait as *mut ()).unwrap();
    let metadata = std::ptr::metadata(trait_object);

    // Create NonNull<dyn SampleTrait> from the data pointer and metadata
    let nonnull_trait_object: NonNull<dyn SampleTrait> = NonNull::from_raw_parts(trait_ptr, metadata);

    unsafe {
        // Ensure trait method and member is preserved
        kani::assert( trait_object.get_value() == nonnull_trait_object.as_ref().get_value(), "trait method and member must correctly preserve");
    }
}
```

using the following command line invocation:
```
cargo kani
```

with Kani version: 0.56.0
Platform: MacOS M2

I expected to see verification results but received the following compilation error:
```
error: Failed to compile `kani_bugs` due to an internal compiler error.: error: internal compiler error: Kani unexpectedly panicked at panicked at cprover_bindings/src/goto_program/stmt.rs:172:9:
                                assertion `left == right` failed: Error: assign statement with unequal types lhs StructTag("tag-_80076268767659083946749461790619121878::FatPtr") rhs Pointer { typ: TypeDef { name: "_80076268767659083946749461790619121878Inner", typ: StructTag("tag-Unit") } }
                                  left: StructTag("tag-_80076268767659083946749461790619121878::FatPtr")
                                 right: Pointer { typ: TypeDef { name: "_80076268767659083946749461790619121878Inner", typ: StructTag("tag-Unit") } }.
```
This issue still persists with assigning `sample_struct` with a concrete value such as:
```rust
let sample_struct = SampleStruct { value: 5 };
```
@zhassan-aws

### PR DESCRIPTION
**Pull request by carolynzech**
Add a match arm for the `AggregateKind::RawPtr(TyKind::RigidTy(RigidTy::Dynamic(..)))` case. 
Pointers to trait objects [are fat](https://github.com/rust-lang/rust/blob/master/library/core/src/ptr/metadata.rs#L20-#L27), so generate a fat pointer for the rvalue.

Resolves #3631

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tests/expected/issue-3631/issue-3631.expected [hunk 1]
```diff
@@ -0,0 +1 @@
+VERIFICATION:- SUCCESS
```
Comment (celinval) [inline tests/expected/issue-3631/issue-3631.expected:1]: Can you please move this test to the `kani` suite? We use expected when we care about validating the output.

**Review by celinval — APPROVED** [review]
Interesting... I believe the case above the one you added doesn't cover slices. But thanks for fixing that.

File: tests/expected/issue-3631/issue-3631.expected (hunk 1 above)
Comment (zhassan-aws) [inline tests/expected/issue-3631/issue-3631.expected:1]:
```suggestion
VERIFICATION:- SUCCESSFUL
```

File: tests/expected/issue-3631/issue-3631.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
```
Comment (zhassan-aws) [inline tests/expected/issue-3631/issue-3631.rs:3]: Please add a comment describing what the test is checking and refer to #3631

File: tests/expected/issue-3631/issue-3631.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+impl SampleTrait for SampleStruct {
+    fn get_value(&self) -> i32 {
+        self.value
+    }
+}
+
+#[cfg(kani)]
+#[kani::proof]
+fn main() {
```
Comment (zhassan-aws) [inline tests/expected/issue-3631/issue-3631.rs:24]: Please rename `main` to something that describes what the harness checks

**Review by zhassan-aws — APPROVED** [review]
Thanks @carolynzech!

**Pull request merged**